frontend/src/pages/ForgotPasswordPage.tsx

```diff
@@ -0,0 +1,212 @@
+import { useState, useEffect, FormEvent, useRef } from 'react';
+import { Alert } from 'react-bootstrap';
+import { Trans, useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import classNames from 'classnames';
+import { ModalContent } from '../components/modals/modalparts/ModalContent';
+import { ModalHeader } from '../components/modals/modalparts/ModalHeader';
+import { Footer } from '../components/Footer';
+import { paths } from '../routers/paths';
+import { Input, errorState } from '../components/forms/FormField';
+import { LoadingSpinner } from '../components/LoadingSpinner';
+import { ReactComponent as MailIcon } from '../icons/mail_icon.svg';
+import { api } from '../api/api';
+import css from './ForgotPassword.module.scss';
+
+const classes = classNames.bind(css);
+
+export const ForgotPasswordPage = () => {
+  const { t } = useTranslation();
+  const defaultDisableSeconds = 30;
+
+  const [linkSent, setLinkSent] = useState(false);
+  const [email, setEmail] = useState('');
+  const [resendDisabled, setResendDisabled] = useState(false);
+  const [resendDisableTime, setResendDisableTime] = useState(
+    defaultDisableSeconds,
+  );
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const confirmationError = errorState(useState(''));
+
+  type Timeout = ReturnType<typeof setTimeout>;
+  const timeout = useRef<Timeout | null>(null);
+  const interval = useRef<Timeout | null>(null);
+
+  // Cleanup timers after unmounting
+  useEffect(
+    () => () => {
+      if (timeout.current !== null) clearTimeout(timeout.current);
+      if (interval.current !== null) clearInterval(interval.current);
+    },
+    [],
+  );
+
+  const sendEmailLink = () => {
+    return api.sendPasswordResetLink(email);
+  };
+
+  const createDisableTimer = () => {
+    setResendDisableTime(defaultDisableSeconds);
+    setResendDisabled(true);
+
+    const countdownInterval = setInterval(() => {
+      setResendDisableTime((previous) => previous - 1);
+    }, 1000);
+
+    setTimeout(() => {
+      clearInterval(countdownInterval);
+      setResendDisabled(false);
+      setResendDisableTime(defaultDisableSeconds);
+    }, defaultDisableSeconds * 1000);
+  };
+
+  const isUniqueError = (err: any) => err.response?.status === 404;
+
+  const handleSend = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    e.stopPropagation();
+    setIsLoading(true);
+
+    try {
+      await sendEmailLink();
+    } catch (err) {
+      if (isUniqueError(err)) {
+        confirmationError.setMessage(t('This email doesn’t exist'));
+      } else {
+        setErrorMessage(t('Email service error'));
+      }
+      setIsLoading(false);
+      return;
+    }
+
+    createDisableTimer();
+    setErrorMessage('');
+    setIsLoading(false);
+    setLinkSent(true);
+  };
+
+  const handleResend = () => {
+    sendEmailLink();
+    createDisableTimer();
```

Comment:
```suggestion
    createDisableTimer(defaultDisableSeconds / 2);
    setErrorMessage('');
    setIsLoading(false);
    setLinkSent(true);
  };

  const handleResend = () => {
    sendEmailLink();
    createDisableTimer(defaultDisableSeconds);
```

The time could still be a parameter, so typo in email isn't quite as long wait